src/passes/OptimizeInstructions.cpp

```diff
@@ -710,6 +710,14 @@ struct OptimizeInstructions
           store->value = unary->value;
         }
       }
+    } else {
+      if (features.hasBulkMemory()) {
+        if (auto* memCopy = curr->dynCast<MemoryCopy>()) {
+          if (auto* ret = optimizeMemoryCopy(memCopy)) {
+            return ret;
+          }
+        }
+      }
```

Comment:
I think this can be simplified to the following:

```suggestion
    } else if (auto* memCopy = curr->dynCast<MemoryCopy>()) {
      assert(features.hasBulkMemory());
      return optimizeMemoryCopy(memCopy));
    }
```